Automattic/jetpack
# Nav redesign / jetpack mu - Unregister customizer items when on block theme.

Fixes # https://github.com/Automattic/dotcom-forge/issues/5853 

## Proposed changes:


Removes the 'customize' menu and adminbar link from atomic sites on block themes running the nav redesign.

* This is essentially a copy of @okmttdhr's work on https://github.com/Automattic/jetpack/pull/36017
* The above linked fix is in the masterbar module, which is disabled as part of the new nav redesign.
* This PR ports the same code to jetpack-mu-wpcom plugin, and applies them to atomic sites. 

BEFORE

<img width="604" alt="Screenshot 2024-03-04 at 2 13 10 PM" src="https://github.com/Automattic/jetpack/assets/28742426/a67afd10-de69-4f3b-9321-3ebf468a86a2">

<img width="353" alt="Screenshot 2024-03-04 at 2 13 00 PM" src="https://github.com/Automattic/jetpack/assets/28742426/5daf304d-9241-4147-8c38-57c4c3b123c1">

AFTER

<img width="570" alt="Screenshot 2024-03-04 at 2 12 26 PM" src="https://github.com/Automattic/jetpack/assets/28742426/c8eb7f0d-348c-4346-976e-b1be08ecebd4">
<img width="356" alt="Screenshot 2024-03-04 at 2 12 34 PM" src="https://github.com/Automattic/jetpack/assets/28742426/3386cf0e-23f2-43b4-a169-f71a419a7922">


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* On a WOA dev blog, enable the "Classic" interface setting under hosting->configuration settings.
* Be proxied, or add the `wpcom_classic_early_release` site option to your site.
* Ensure you have a block theme enabled.
* Build and sync this branch of jetpack-mu-wpcom. PCYsg-Osp-p2
* Verify the "Customizer" link no longer appears under:
   *  "Appearance" in the wp-admin left nav menu
   * In the top adminbar while viewing the front end of the site.
* visit the customizer directly by URL `*/wp-admin/customize.php`, verify this loads as expected without errors.

* Enable a classic theme and verify the customize menu reappears.
